contracts/dnsregistrar/OffchainDNSResolver.sol

```diff
@@ -199,10 +212,16 @@ contract OffchainDNSResolver is IExtendedResolver, IERC165 {
         uint256 startIdx,
         uint256 lastIdx
     ) internal pure returns (bytes memory) {
-        // TODO: Concatenate multiple text fields
-        uint256 fieldLength = data.readUint8(startIdx);
-        assert(startIdx + fieldLength < lastIdx);
-        return data.substring(startIdx + 1, fieldLength);
+        bytes memory result = new bytes(0);
+        uint256 idx = startIdx;
+        while (idx < lastIdx) {
+            uint256 fieldLength = data.readUint8(idx);
+            assert(idx + fieldLength + 1 <= lastIdx);
+            bytes memory field = data.substring(idx + 1, fieldLength);
+            result = abi.encodePacked(result, field);
```

Comment:
Iterated concatenation like this is `O(n^2)`. Instead, do a first loop to get the total length of the string, then allocate a new string and copy them into it in a second loop. You will probably need to write a `strcpy` method in `BytesUtils` that uses the existing `memcpy` implementation.